Realm/RLMQueryUtil.mm

```diff
@@ -401,17 +401,12 @@ void add_bool_constraint_to_query(realm::Query &query, NSPredicateOperatorType o
     }
 }
 
-template <typename T>
-void add_string_constraint_to_query(realm::Query &query,
-                                    NSPredicateOperatorType operatorType,
-                                    NSComparisonPredicateOptions predicateOptions,
-                                    Columns<String> &&column,
-                                    T value) {
-    bool caseSensitive = !(predicateOptions & NSCaseInsensitivePredicateOption);
-    bool diacriticInsensitive = (predicateOptions & NSDiacriticInsensitivePredicateOption);
-    RLMPrecondition(!diacriticInsensitive, @"Invalid predicate option",
-                    @"NSDiacriticInsensitivePredicateOption not supported for string type");
 
+struct SkipCheckForEmptyAndNullStrings {};
+
+template <typename T>
+void add_substring_constraint_to_query(SkipCheckForEmptyAndNullStrings, Query& query, NSPredicateOperatorType operatorType, bool caseSensitive, Columns<String> column, T value)
```

Comment:
Leftovers from an earlier approach that I didn't simplify sufficiently. Will fix.